layers/fully_connected.py

```diff
@@ -197,25 +197,26 @@ def get_eps(self, size):
         eps = Variable(eps)
         return eps
 
+    def sample_pW(self):
+        return self.floatTensor(self.in_features, self.out_features).normal_()
+
+    def sample_pb(self):
+        return self.floatTensor(self.out_features).normal_()
+
     def sample_W(self):
-        W = self.mean_w
-        if self.training:
-            eps = self.get_eps(self.mean_w.size())
-            W = W.add(eps.mul(self.logvar_w.mul(0.5).exp_()))
```

Comment:
Any specific reason on why you changed this? Usually, for prediction we use the deterministic equivalent, i.e. the mean of the distribution.